File: implementation/scipamato/scipamato-core-entity/src/test/java/ch/difty/scipamato/core/entity/newsletter/NewsletterTopicFilterTest.java
package ch.difty.scipamato.core.entity.newsletter;

import static org.assertj.core.api.Assertions.assertThat;

import nl.jqno.equalsverifier.EqualsVerifier;
import nl.jqno.equalsverifier.Warning;
import org.junit.jupiter.api.Test;

public class NewsletterTopicFilterTest {

    private final NewsletterTopicFilter f = new NewsletterTopicFilter();

    @Test
    public void getAndSet() {
        f.setTitleMask("titleMask");

        assertThat(f.getTitleMask()).isEqualTo("titleMask");

        assertThat(f.toString()).isEqualTo("NewsletterTopicFilter(titleMask=titleMask)");
    }

    @Test
    public void equals() {
        EqualsVerifier
            .forClass(NewsletterTopicFilter.class)
            .withRedefinedSuperclass()
            .suppress(Warning.STRICT_INHERITANCE, Warning.NONFINAL_FIELDS)
            .verify();
    }

    @Test
    public void assertEnumFields() {
        assertThat(NewsletterTopicFilter.NewsletterTopicFilterFields.values())
            .extracting("name")
            .containsExactly("titleMask");
    }

}

File: core/core-entity/src/test/kotlin/ch/difty/scipamato/core/entity/newsletter/NewsletterTopicFilterTest.kt
package ch.difty.scipamato.core.entity.newsletter

import nl.jqno.equalsverifier.EqualsVerifier
import org.amshove.kluent.shouldBeEqualTo
import org.amshove.kluent.shouldContainSame
import org.junit.jupiter.api.Test

internal class NewsletterTopicFilterTest {

    private val f = NewsletterTopicFilter()

    @Test
    fun getAndSet() {
        f.titleMask = "titleMask"
        f.titleMask shouldBeEqualTo "titleMask"
        f.toString() shouldBeEqualTo "NewsletterTopicFilter(titleMask=titleMask)"
    }

    @Test
    fun equals() {
        EqualsVerifier.simple()
            .forClass(NewsletterTopicFilter::class.java)
            .withRedefinedSuperclass()
            .verify()
    }

    @Test
    fun assertEnumFields() {
        NewsletterTopicFilter.NewsletterTopicFilterFields.entries.map { it.fieldName } shouldContainSame
            listOf("titleMask")
    }
}
